src/components/ImmutableDevices/helpers.js

```diff
@@ -0,0 +1,57 @@
+import {
+  CheckCircleIcon,
+  CubeIcon,
+  ExclamationCircleIcon,
+  ExclamationTriangleIcon,
+  InProgressIcon,
+  ModuleIcon,
+  PlusCircleIcon,
+  QuestionCircleIcon,
+  RepositoryIcon,
+  SearchIcon,
+  TimesCircleIcon,
+  UnknownIcon,
+} from '@patternfly/react-icons';
+import dangerColor from '@patternfly/react-tokens/dist/esm/global_danger_color_100';
+import warningColor from '@patternfly/react-tokens/dist/esm/global_warning_color_100';
+import successColor from '@patternfly/react-tokens/dist/esm/global_success_color_100';
+import infoColor from '@patternfly/react-tokens/dist/esm/global_info_color_100';
+import activeColor from '@patternfly/react-tokens/dist/esm/global_active_color_100';
+
+export const colorMapper = {
+  green: successColor.value,
+  yellow: warningColor.value,
+  lightBlue: infoColor.value,
+  blue: activeColor.value,
+  red: dangerColor.value,
+};
+
+export const iconMapper = {
+  unknown: UnknownIcon,
+  repository: RepositoryIcon,
+  search: SearchIcon,
+  module: ModuleIcon,
+  cube: CubeIcon,
+  question: QuestionCircleIcon,
+  plus: PlusCircleIcon,
+  checkCircle: CheckCircleIcon,
+  exclamationTriangle: ExclamationTriangleIcon,
+  timesCircle: TimesCircleIcon,
+  inProgress: InProgressIcon,
+  exclamationCircle: ExclamationCircleIcon,
+};
+
+export const getDeviceStatus = (
+  deviceStatus,
+  isUpdateAvailable,
+  dispatcherStatus
+) =>
+  dispatcherStatus === 'ERROR'
+    ? 'error'
+    : dispatcherStatus === 'UNRESPONSIVE'
+    ? 'unresponsive'
+    : deviceStatus === 'UPDATING'
+    ? 'updating'
+    : isUpdateAvailable
+    ? 'updateAvailable'
+    : 'upToDate';
```

Comment:
There is no way this is humanly readable. It might be better to use proper `if()...else` here, just to allow for better reading what is going on.

Maybe there is even a better way to match these device states?

Reply:
Thank you for the proactive review. This is yet in Draft state. PTAL on this ready [PR](https://github.com/RedHatInsights/insights-inventory-frontend/pull/2030#event-10524769575) if you have free capacity.